### PR TITLE
Reworked cleaning up API test files

### PR DESCRIPTION
**Pull request by brtnfld**
Only removing test files if exist to account for skipped tests

### CODE REVIEW

File: test/API/H5_api_test.c [hunk 1]
```diff
@@ -263,7 +263,9 @@ main(int argc, char **argv)
     H5_api_test_run();
 
     printf("Cleaning up testing files\n");
-    H5Fdelete(H5_api_test_filename, fapl_id);
+    if (HDaccess(H5_api_test_filename, F_OK) != -1) {
```
Comment (jhendersonHDF) [inline test/API/H5_api_test.c]: Probably need to use H5Fis_accessible everywhere instead. `access` won't work where there's not a physical file.

Comment (brtnfld) [reply test/API/H5_api_test.c]: fixed

File: test/API/H5_api_object_test.c [hunk 2]
```diff
@@ -7375,6 +7365,23 @@ object_visit_noop_callback(hid_t o_id, const char *name, const H5O_info2_t *obje
...
+ * Cleanup temporary test files
+ */
+static void
+cleanup_files(void)
+{
+    char filename[H5_API_TEST_FILENAME_MAX_LENGTH];
+
+    snprintf(filename, H5_API_TEST_FILENAME_MAX_LENGTH, "%s%s", test_path_prefix,
+             OBJECT_COPY_BETWEEN_FILES_TEST_FILE_NAME);
+    remove_test_file(NULL, filename);
```
Comment (jhendersonHDF) [inline test/API/H5_api_object_test.c]: Could also just pass `test_path_prefix` to `remove_test_file` here to be even simpler.

**Review by jhendersonHDF — APPROVED** [review]

File: test/API/H5_api_object_test.c (hunk 2 above)
Comment (brtnfld) [reply test/API/H5_api_object_test.c]: updated

**Review by vchoi-hdfgroup — APPROVED** [review]

**Review by derobins — APPROVED** [review]

**Pull request merged**